src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -103,20 +120,91 @@ def _check_if_empty_ts(time_step: BindingConstraintFrequency, time_series: Optio
-        }
-        self.ini_file.ini_dict = binding_constraints_ini_content
+    def _write_binding_constraint_ini(
+        self,
+        properties: BindingConstraintProperties,
+        constraint_name: str,
+        constraint_id: str,
+        terms: Optional[list[ConstraintTerm]] = None,
+    ) -> None:
+        """
+        Write or update a binding constraint in the INI file.
+
+        """
+
+        current_ini_content = self.ini_file.ini_dict_binding_constraints or {}
+
+        existing_section = next(
+            (section for section, values in current_ini_content.items() if values.get("name") == constraint_name),
+            None,
+        )
+
+        if existing_section:
+            existing_terms = current_ini_content[existing_section]
+
+            serialized_terms = (
+                {term.id: serialize_term_data(term.data, term.offset, term.weight) for term in terms} if terms else {}
+            )
+
+            existing_terms.update(serialized_terms)  # type: ignore
+            current_ini_content[existing_section] = existing_terms
+
+            # Persist the updated INI content
+            self.ini_file.ini_dict_binding_constraints = current_ini_content
+            self.ini_file.write_ini_file()
+        else:
+            terms_dict = (
+                {
+                    term.id: ConstraintTerm(data=term.data, offset=term.offset, weight=term.weight)
+                    if isinstance(term.data, (LinkData, ClusterData))
+                    else term
+                    for term in terms
+                }
+                if terms
+                else {}
+            )
+
+            full_properties = BindingConstraintPropertiesLocal(
+                constraint_name=constraint_name,
+                constraint_id=constraint_id,
+                terms=terms_dict,
+                **properties.model_dump(),
+            )
+
+            section_index = len(current_ini_content)
+            current_ini_content[str(section_index)] = full_properties.list_ini_fields
+
+        self.ini_file.ini_dict_binding_constraints = current_ini_content
         self.ini_file.write_ini_file()
 
     def add_constraint_terms(self, constraint: BindingConstraint, terms: list[ConstraintTerm]) -> list[ConstraintTerm]:
-        new_terms = constraint.local_properties.terms | {
-            term.id: term for term in terms if term.id not in constraint.get_terms()
+        """
+        Add terms to a binding constraint and update the INI file.
+
+        Args:
+            constraint (BindingConstraint): The binding constraint to update.
+            terms (list[ConstraintTerm]): A list of new terms to add.
+
+        Returns:
+            list[ConstraintTerm]: The updated list of terms.
+        """
+
+        new_terms = {
+            **constraint.local_properties.terms,
+            **{term.id: term for term in terms if term.id not in constraint.get_terms()},
```

Comment:
Shouldn't we raise if we want to add an already existing term instead of just not writing it ? Don't know

Reply:
Not resolved :/. I think we should raise